packages/flutter_tools/lib/src/device.dart

```diff
@@ -177,7 +177,8 @@ abstract class Device {
   /// [platformArgs] allows callers to pass platform-specific arguments to the
   /// start call.
   Future<LaunchResult> startApp(
-    ApplicationPackage package, {
+    ApplicationPackage package,
+    BuildMode mode, {
```

Comment:
Can you add a note to the method dartdocs that `[mode]` is not used by all implementations?